### DISCUSSION TITLE
Refetching "admin._index.ts" via "fetcher.load" from inside "admin._index.ts" loads loader in "admin.ts"

### DISCUSSION THREAD

**Discussion by GeniusChinez** [remix-run/remix#7412] [Q&A] [answered]
In my remix.run app, I am using v2 routing conventions, and am facing an issue. I have a file named "admin._index.ts", which is the admin dashboard route, and contains a loader and a component. I also have another file named "admin.ts" (root for all admin routes) and also contains a loader and a component. Now, when im in "admin._index.ts" and make a get request to "/admin" (to refetch via fetcher.load), the loader in "admin.ts" is executed instead of the loader in the current file "admin._index.ts". Is there a way to resolve this without renaming "admin._index.ts" to "admin.dashboard.ts" and redirecting from "admin._index.ts" to "admin.dashboard.ts"?

**Comment by kiliman** [comment, marked as answer]
`fetcher.load()` is intended to be used on resource routes, not UI routes.

For UI routes, you should be using `useLoaderData` and `useRevalidator` to refresh the loader.

If you must, then use `fetcher.load('/admin?index')`. This tells Remix to call the `index` route, since both routes are accessible from `/admin`.

https://remix.run/docs/en/1.19.3/guides/routing#what-is-the-index-query-param

**Reply by GeniusChinez** [reply]
You're the man !!! Thank you.
